units/crons.go

```diff
@@ -322,8 +322,13 @@ func PopulateIdleHostJobs(env evergreen.Environment) amboy.QueueOperation {
 			for _, d := range distrosFound {
 				distroIDsFound = append(distroIDsFound, d.Id)
 			}
-			invalidDistroIDs := util.GetSetDifference(distroIDsToFind, distroIDsFound)
-			return fmt.Errorf("distro ids %s not found", strings.Join(invalidDistroIDs, ","))
+			missingDistroIDs := util.GetSetDifference(distroIDsToFind, distroIDsFound)
+			for _, distroID := range missingDistroIDs {
+				catcher.Wrapf(host.DecommissionHostsWithDistroId(distroID), "error marking hosts in distro '%s' as decommissioned", distroID)
```

Comment:
This won't log to the event log, but would be nice if it did

Reply:
I'd be fine with splunk

Reply:
Done. I changed this to find the hosts in the missing distros and call `SetDecommissioned()` for the event logging.